### PR TITLE
Add reverted aria audit rules for dev toolbar

### PR DESCRIPTION
**Pull request by bluwy**
## Changes

Un-revert https://github.com/withastro/astro/pull/9295 and https://github.com/withastro/astro/pull/9293

The main change is adding `optimizeDeps.include: ['astro > aria-query', 'astro > axobject-query']` to fix it.

## Testing

Tested manually with `pnpm link`, and `pnpm pack` + install locally (to emulate when it's published)

## Docs

It doesn't seem like we document the audit rules today.

### CODE REVIEW

**Review by Princesseuh — APPROVED** [review]

File: packages/astro/src/runtime/client/dev-overlay/plugins/audit/a11y.ts [hunk 1]
```diff
@@ -489,3 +563,81 @@ const a11y_non_interactive_element_to_interactive_role_exceptions = {
 	td: ['gridcell'],
 	fieldset: ['radiogroup', 'presentation'],
 };
+
+const combobox_if_list = ['email', 'search', 'tel', 'text', 'url'];
```
Comment (ematipico) [inline packages/astro/src/runtime/client/dev-overlay/plugins/audit/a11y.ts:567]: Can we change snake case names to camel case?

**Review by ematipico — APPROVED** [review]
Looks good. I would update the variable names, and use camel case, since that's what we use in the code base

File: packages/astro/src/runtime/client/dev-overlay/plugins/audit/a11y.ts (hunk 1 above)
Comment (bluwy) [reply packages/astro/src/runtime/client/dev-overlay/plugins/audit/a11y.ts:567]: I can change them to camelCase. It was snake_case as it's borrowed from Svelte and that's the convention they use.

Comment (bluwy) [reply packages/astro/src/runtime/client/dev-overlay/plugins/audit/a11y.ts:567]: Seems like there's a lot that uses snake_case. I think maybe we should do one PR later to convert them all at once.

**Review by sarah11918 — APPROVED** [review]
Docs is happy!  🥳

**Pull request merged**